### PR TITLE
XRDDEV-1425 limit HSM CSR creation to SIGNING usage

### PR DESCRIPTION
**Pull request by jansu76**
https://jira.niis.org/browse/XRDDEV-1425

### CODE REVIEW

File: src/proxy-ui-api/frontend/src/store/modules/certificateSignRequest.ts [hunk 1]
```diff
@@ -103,6 +106,10 @@ export const csrGetters: GetterTree<CsrState, RootState> = {
     return state.keyLabel;
   },
   isUsageReadOnly(state): boolean {
+    // if creating CSR for a hardware token, only sign CSRs can be created
+    if (state.tokenType == TokenType.HARDWARE) {
```
Comment (TJaakkola) [inline src/proxy-ui-api/frontend/src/store/modules/certificateSignRequest.ts:110]: In this case it doesn't matter but it's a good practise to always use three equal signs when doing a comparison in javascript.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality

For example. 

1 == '1' // true 
1 === '1' // false

File: src/proxy-ui-api/frontend/src/store/modules/certificateSignRequest.ts [hunk 2]
```diff
@@ -194,6 +204,13 @@ export const actions: ActionTree<CsrState, RootState> = {
   setCsrTokenId({ commit }, tokenId: string) {
     commit('storeCsrTokenId', tokenId);
   },
+  setCsrTokenType({ commit }, tokenType: string) {
+    if (tokenType == TokenType.HARDWARE) {
```
Comment (TJaakkola) [inline src/proxy-ui-api/frontend/src/store/modules/certificateSignRequest.ts:208]: Strict equality like above

**Review by TJaakkola — CHANGES_REQUESTED** [review]
A couple of minor changes required

**Review by TJaakkola — APPROVED** [review]

File: src/proxy-ui-api/frontend/src/store/modules/certificateSignRequest.ts (hunk 2 above)
Comment (jansu76) [reply src/proxy-ui-api/frontend/src/store/modules/certificateSignRequest.ts]: fixed

File: src/proxy-ui-api/frontend/src/store/modules/certificateSignRequest.ts (hunk 1 above)
Comment (jansu76) [reply src/proxy-ui-api/frontend/src/store/modules/certificateSignRequest.ts]: fixed

**Pull request merged**